### DISCUSSION TITLE
Bug using openAccountModal() hook

### DISCUSSION THREAD

**Discussion by Lakszy** [rainbow-me/rainbowkit#1810] [Q&A] [answered]
I am not able to open the modal for all the networks 
It's only opening for the mantle chain network and I am unable to figure out the error
```
import { ConnectButton } from '@rainbow-me/rainbowkit';
import '../App.css';
import { useAccount } from 'wagmi';

function ConnectBtn() {
  const { address, isConnected } = useAccount();



  return (
    <ConnectButton.Custom>
      {({
        account,
        chain,
        openAccountModal,
        openChainModal,
        openConnectModal,
        authenticationStatus,
        mounted,
      }) => {
        const ready = mounted && authenticationStatus !== 'loading';
        const connected =
          ready &&
          account &&
          chain &&
          (!authenticationStatus ||
            authenticationStatus === 'authenticated');

        return (
          <div
            {...(!ready && {
              'aria-hidden': true,
              'style': {
                opacity: 0,
                pointerEvents: 'none',
                userSelect: 'none',
              },
            })}
          >
            {(() => {
              if (!connected) {
                return (
                  <button className='btn btn-riv-primary whitespace-nowrap' onClick={openConnectModal} type="button">
                    Connect Wallet
                  </button>
                );
              }

              return (
                <div className='flex_btn_menu_connect'>
                  <div className='flexTwoBtn'>
                    <button className='btn btn-riv-secondary connect_btn_wdth imDark mrMobile' onClick={openAccountModal} type="button">
                      <div className='flexConnec'>
                        <h6 className=' imBalance whitespace-nowrap'>{account.displayName}</h6>
                        <h6 className='imBalance whitespace-nowrap'>
                          {account.displayBalance
                            ? ` (${account.displayBalance})`
                            : ''}
                        </h6>
                      </div>
                    </button>

                  </div>
                </div>
              );
            })()}
          </div>
        );
      }}
    </ConnectButton.Custom>
  );
}

export default ConnectBtn;

```

**Comment by Lakszy** [comment]
@KosmosKey @markdalgleish

Help would be much appreciated
Thanks!

**Reply by 0xmkh** [reply]
Could you please share your RainbowKit and Wagmi setup ?

**Reply by Lakszy** [reply]
Thank you for your response. I just found my mistake. While configuring the chain, I was not passing all custom-defined networks.
![image](https://github.com/rainbow-me/rainbowkit/assets/117038630/b92ca080-11c4-4333-a4c5-730ef24d4cf6)

**Reply by Lakszy** [reply]
Other than that, I am still figuring out if I can customize the openAccountModal yet or not. Do you have any suggestions on that?

@KosmosKey 

- [ ![image](https://github.com/rainbow-me/rainbowkit/assets/117038630/19c8098a-94bd-41bf-a08f-0b45c5e83f10)] 

- [x] ![image](https://github.com/rainbow-me/rainbowkit/assets/117038630/3168b371-5dbd-45e8-93ec-e9951bb8bb47)

**Reply by 0xmkh** [reply]
@Lakszy Glad it worked! Regarding to the custom `openAccountModal`, yes you can customise how you're calling it. This means you don't have to use `<ConnectButton.Custom>`, but rather use our hooks directly like shown here: https://www.rainbowkit.com/docs/modal-hooks

Let me know if that helps 🙏

**Comment by Lakszy** [comment]
![image](https://github.com/rainbow-me/rainbowkit/assets/117038630/a282e332-285f-4423-afcb-91acc4006feb)
@KosmosKey Thank you for responding.
It shows '0 MNT' if I want to hide this or add some custom text, like a name or something.
Is this possible to do right now?

**Reply by 0xmkh** [reply, marked as answer]
@Lakszy It's unfortunately not possible :/

You can always create your own modal.

**Reply by Lakszy** [reply]
Thanks for the help dude. 
🙏🏻
